src/correlator_bosonic_recursive.jl

```diff
@@ -167,40 +179,81 @@ function add_operator_fermi(
       L_prev = (op_ind > 1 ? delta(dag(ln[op_ind - 1])', ln[op_ind - 1]) : 1.0) #initialize left environment
     end
 
-    L = L_prev#copy(L_prev) #copy cached environment after applying previous operator (this can be simplified)
+    L = L_prev  #copy(L_prev) #copy cached environment after applying previous operator (this can be simplified)
     op_psi = psi[op_ind]
 
     if jw % 2 != 0
       op_psi = apply(op("F", s[op_ind]), op_psi) #apply jordan wigner string if needed
     end
-    op_psi = apply(op(ops[counter], s[op_ind]), op_psi)  #apply operator in the spot #counter
-
-    jw_next = 0
-    if ops[counter] == "Cdagup" ||
-      ops[counter] == "Cdag" ||
-      ops[counter] == "Cup" ||
-      ops[counter] == "C"
-      jw_next = jw + 1
-      op_psi = apply(op("F", s[op_ind]), op_psi) #track if a fermionic operator was applied
-    elseif ops[counter] == "Cdagdn" || ops[counter] == "Cdn"
-      jw_next = jw + 1
-      op_psi = apply(op("F", s[op_ind]), op_psi) #for spin down operator we need a j-w term on-sitw
+
+    for i in 0:repeat
+      op_psi = apply(op(ops[perm_ind[counter + repeat - i]], s[op_ind]), op_psi)
+    end
+
+    jw_next = jw
+
+    for i in 0:repeat
+      if has_fermion_string(ops[perm_ind[counter + repeat - i]], s[op_ind])
+        jw_next = jw_next + 1
+        op_psi = apply(op("F", s[op_ind]), op_psi) #track if a fermionic operator was applied
+      end
     end
 
     L = L * op_psi * psi_dag[op_ind]  #generate left environment to store
 
-    if counter == N
+    if counter + repeat == N
       R = ((op_ind) < length(psi) ? delta(dag(ln[op_ind]), ln[op_ind]') : ITensor(1.0)) #create right system
-      C[tuple([element[k] for k in [findall(x -> x == j, indices)[1] for j in TupleTools.sort(indices)]]...)] = inner(
-        dag(L), R
-      )
-      #push!(C, tuple([element[k] for k in [findall(x->x==j,indices)[1] for j in sort(indices)]]...) => inner(dag(L), R))
+
+      # re-arranging the sorted element list using permutations
+      perm_elem = element[sortperm(perm_ind)]
+
+      # checking for fermion operators and keeping track of anti-commutations
+      ferm_sites =
+        Int64.(perm_elem[findall(x -> has_fermion_string(x, s[op_ind]), ops)])
```

Comment:
```suggestion
        Int.(perm_elem[findall(x -> has_fermion_string(x, s[op_ind]), ops)])
```
That's more generic to 32-bit vs. 64-bit systems.